packages/opentelemetry-plugin-grpc-js/src/server/serverStreamAndBidi.ts

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Span, CanonicalCode } from '@opentelemetry/api';
+import { RpcAttribute } from '@opentelemetry/semantic-conventions';
+import type * as grpcJs from '@grpc/grpc-js';
+import type { GrpcJsPlugin } from '../grpcJs';
+import { GrpcEmitter } from '../types';
+import { CALL_SPAN_ENDED, grpcStatusCodeToCanonicalCode } from '../utils';
+
+/**
+ * Handle patching for serverStream and Bidi type server handlers
+ */
+export function serverStreamAndBidiHandler<RequestType, ResponseType>(
+  plugin: GrpcJsPlugin,
+  span: Span,
+  call: GrpcEmitter,
+  original:
+    | grpcJs.handleBidiStreamingCall<RequestType, ResponseType>
+    | grpcJs.handleServerStreamingCall<RequestType, ResponseType>
+): void {
+  let spanEnded = false;
+  const endSpan = () => {
+    if (!spanEnded) {
```

Comment:
I'm thinking that we should expose a property on the SDK's span that represent if the span has been ended. There are few plugins that implement something like this which add memory overhead for nothing, WDYT ?